code/modules/antagonists/eldritch_cult/eldritch_effects.dm

```diff
@@ -0,0 +1,218 @@
+	var/datum/antagonist/heretic/cultie = user.mind.has_antag_datum(/datum/antagonist/heretic)
+	var/list/knowledge = cultie.get_all_knowledge()
+	var/list/atoms_in_range = list()
+
+	for(var/atom/atom_in_range in range(1,src))
+		if(istype(atom_in_range,/mob/living))
+			var/mob/living/living_in_range = atom_in_range
+			if(living_in_range.stat != DEAD || living_in_range == user) // we only accept corpses, no living beings allowed.
+				continue
+		atoms_in_range += atom_in_range
+
+	for(var/X in knowledge)
+		var/datum/eldritch_knowledge/current_eldritch_knowledge = X
+		var/list/local_required_atoms = current_eldritch_knowledge.required_atoms
+		if(!local_required_atoms || local_required_atoms.len == 0)
+			continue
+
+		var/list/selected_atoms = list()
+
+		if(!current_eldritch_knowledge.recipe_snowflake_check(atoms_in_range,drop_location(),selected_atoms))
+			continue
+
+		for(var/local_required_atom in local_required_atoms)
+			local_required_atoms[local_required_atom] = FALSE
+
+		for(var/local_atom_in_range in atoms_in_range)
+			var/atom/first_atom = local_atom_in_range
+			for(var/local_required_atom_in_range in local_required_atoms)
+				var/list/local_required_atom_list_in_range = local_required_atom_in_range
+				if(!is_type_in_list(first_atom,local_required_atom_list_in_range) || local_required_atoms[local_required_atom_in_range] == TRUE)
+					continue
+				local_required_atoms[local_required_atom_in_range] = TRUE
+				selected_atoms |= first_atom
+
+		if(selected_atoms.len == 0)
+			continue
+		var/skip = FALSE
+		for(var/secondary_local_required_atom in local_required_atoms)
+			if(local_required_atoms[secondary_local_required_atom] != TRUE)
+				skip = TRUE
+				break
+
+		if(skip)
+			continue
+		flick("[icon_state]_active",src)
+		playsound(user, 'sound/magic/castsummon.ogg', 75, TRUE)
+		current_eldritch_knowledge.on_finished_recipe(user,selected_atoms,loc)
+		current_eldritch_knowledge.cleanup_atoms(selected_atoms)
+		return
+	to_chat(user,"<span class='warning'>Your ritual failed! You used either wrong components or are missing something important!</span>")
+
+/obj/effect/eldritch/big
+	name = "Transmutation rune"
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "eldritch_rune1"
+	pixel_x = -32 //So the big ol' 96x96 sprite shows up right
+	pixel_y = -32
+
+/**
+  * #Reality smash tracker
+  *
+  * Stupid fucking list holder, DONT create new ones, it will break the game, this is automnatically created whenever eldritch cultists are created.
+  *
+  * Tracks relevant data, generates relevant data, useful tool
+  */
+/datum/reality_smash_tracker
+	///list of tracked reality smashes
+	var/list/smashes = list()
+	///List of mobs with ability to see the smashes
+	var/list/targets = list()
+
+/datum/reality_smash_tracker/New()
+	. = ..()
+	if(!GLOB.reality_smash_track)
+		GLOB.reality_smash_track = src
+	else
+		stack_trace("/datum/reality_smash_tracker was initialized while one already exists. Deleting")
```

Comment:
If this is global why not just initialize it there or make a getter.

Reply:
done